logstash-core/src/main/java/org/logstash/instrument/witness/Witness.java

```diff
@@ -28,8 +30,10 @@
     private final ReloadWitness reloadWitness;
     private final EventsWitness eventsWitness;
     private final PipelinesWitness pipelinesWitness;
+    private final ProcessWitness processWitness;
```

Comment:
@jakelandis this one could probably just be made `static` right? It's not gonna change during the lifetime of the JVM is it?

Reply:
It is an instance inside a singleton [1], so the cost difference is trival.

[1]
 almost a singleton. The Ruby lifecyle and JVM lifecycles are not in sync for the tests, so for testing, keeping this an instance of this class makes life simpler.

Reply:
@jakelandis they aren't? Where + why? (having them not in sync seems like a big problem actually, much of our functionality around `RubyUtils` relies on the fact that we can cache the global `Ruby` instance)

Reply:
@original-brownbear - I should re-phrase, the Ruby **Agent** life-cycle and JVM lifecycle are not sync for the rspec tests.  Specifically it is in the rspec unit tests, we create hundreds of new agents, on the same JVM.  Outside of testing, there is a 1:1 relationship.

Reply:
@jakelandis hmm hmm hmm it still seems dirty to have something related to the process, not `static final` but I don't really know enough about the details here to suggest a better approach at this point :) => Let's just go with this for now, it's a really minor point anyways => sorry for wasting time ;)